### ISSUE TITLE
Documentation confusing on whether SSD and RetinaNet count background as class object

### ISSUE THREAD

**Issue by douglasrizzo** [pytorch/vision#4106]
The documentation for the `SSD` class mentions that we should not count the background as an object class when passing the number of classes as a parameter to instantiate an SSD object.

https://github.com/pytorch/vision/blob/959666891589c35e6d225943253f523cffbae4cc/torchvision/models/detection/ssd.py#L144

However, further down in the same file, an SSD object is instantiated in a function that explicitly says that the background **should** be counted as an object class, but this is not taken into account in the code (i.e. I did not see `num_classes` be decremented by one when creating the SSD object).

https://github.com/pytorch/vision/blob/959666891589c35e6d225943253f523cffbae4cc/torchvision/models/detection/ssd.py#L589

Here is the documentation for this function, which says we should include the background in the number of classes.

https://github.com/pytorch/vision/blob/959666891589c35e6d225943253f523cffbae4cc/torchvision/models/detection/ssd.py#L563

This is confusing. Should we or should we not count the background as an object class when instantiating the SSD? In either case, how should object classes be ID'd during training?

As an example, with Faster RCNN, the background is counted as an object class (with ID 0 reserved for it) and actual object classes are identified during training starting from ID 1. What should be the procedure for SSD?

I have also opened [a topic](https://discuss.pytorch.org/t/do-we-have-to-count-the-background-as-an-object-class-for-ssd/124927) in the forums, since this is both a personal question of mine as well as a possible issue in the docs (or the code).

**Comment by datumbox** [comment]
@douglasrizzo Thanks for reporting. 

Blame me for copy-pasting the doc string of RetinaNet which had exactly the same issue. The `num_classes` should include the background. I've issued a PR that fixes the problem.

**Comment by douglasrizzo** [comment]
@datumbox Thanks for the reply. I actually believe the RetinaNet documentation suffers from a similar problem.

The documentation for the `RetinaNet` class tells us to **exclude** the background as an object class:

https://github.com/pytorch/vision/blob/1962fddb116f697d7059fd11c2f204839460b214/torchvision/models/detection/retinanet.py#L259

The function `retinanet_resnet50_fpn` tells us to **include** the background as an object class:

https://github.com/pytorch/vision/blob/1962fddb116f697d7059fd11c2f204839460b214/torchvision/models/detection/retinanet.py#L608

But when a `RetinaNet` object is created inside of `retinanet_resnet50_fpn`, the number of classes passed to `retinanet_resnet50_fpn` (which counts the background) is not decremented by 1 before being used in `RetinaNet`:

https://github.com/pytorch/vision/blob/1962fddb116f697d7059fd11c2f204839460b214/torchvision/models/detection/retinanet.py#L622

Since `num_classes` is only used inside `retinanet_resnet50_fpn` to instantiate a `RetinaNet` object, my bet is that the documentation for `retinanet_resnet50_fpn` is wrong.

**Comment by douglasrizzo** [comment]
I just saw your PR fixes both documents.

**Comment by WZMIAOMIAO** [comment]
@datumbox 
hi, I think num_classes in retinanet should exclude the background.
because in classification loss using ```binary_cross_entropy_with_logits```, don't need to consider background.

**Comment by Xonxt** [comment]
Sorry to reopen the issue, but the other question was not answered: 
how should the labels be numbered during training (for custom datasets) - starting from 0 or from 1? 
And then during inference, does the model return the labels starting from 0 or from 1?
